### PR TITLE
Increase paid storage and VDF revelation operations added

### PR DESCRIPTION
**Pull request by AbineshECAD**

### CODE REVIEW

**Comment by AbineshECAD** [comment]
Issue: https://github.com/ecadlabs/signatory/issues/237

File: pkg/tezos/operation.go [hunk 1]
```diff
@@ -1512,3 +1552,69 @@ func parseEntrypoint(buf *[]byte) (e string, err error) {
...
+}
+
+// decodeStorageBytes decodes a zarith encoded integer from the entire input byte array.
+func decodeStorageBytes(bamt []byte) (*big.Int, error) {
+	if len(bamt) == 0 {
+		return nil, fmt.Errorf("expected non-empty byte array")
+	}
+
+	// Split input into 8-bit bitstrings
+	segments := make([]string, len(bamt))
```
Comment (e-asphyx) [inline pkg/tezos/operation.go:1580]: Please don't do everything via string conversions

**Comment by danielelisi** [comment]
Tested the `increase_paid_storage` operation with the following `tezos-client` command

```
$ tezos-client increase the paid storage of <contract_address> by 5 bytes from <wallet>
```


Signatory logs
```
INFO[0009] GET /authorized_keys                          duration="160.519µs" hostname="localhost:6732" method=GET path=/authorized_keys start_time="2022-09-27T16:39:29-07:00" status=200
INFO[0009] Requesting signing operation                  kind="[increase_paid_storage]" op=generic pkh=tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH vault=File vault_name=local_file_keys
INFO[0009] About to sign raw bytes                       kind="[increase_paid_storage]" op=generic pkh=tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH raw=03811184220559e738f28553d88389f5334fa98516a55b5ecfd4c9be8d4ce4c55e7100bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b6f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b00 vault=File vault_name=local_file_keys
INFO[0009] Signed generic successfully                   kind="[increase_paid_storage]" op=generic pkh=tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH vault=File vault_name=local_file_keys
INFO[0009] POST /keys/tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH  duration="754.011µs" hostname="localhost:6732" method=POST path=/keys/tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH start_time="2022-09-27T16:39:29-07:00" status=200
```

**Comment by danielelisi** [comment]
Since `tezos-client` doesn't have a command for `vdf_revelation` operations we can instead craft it manually.

The `vdf_revelation` expects an operation with tag number `08` followed by a 200 bytes string.

I took the time to better understand the parsing logic Signatory uses:
 - the first byte `03` indicates the request type. In this case, it's a `generic` request.
 - The following 32 bytes `baeb51250afe4344ed1081b8df92a8e8f7f7286b6ed82585fedc7828fd0c1312` reference the block hash
 - the following byte `08` indicates the operation type which in this case is mapped to `vdf_revelation`
 - the following 200 bytes are the content of the `vdf_revelation`

Here is the raw operation I used 
```
0x03baeb51250afe4344ed1081b8df92a8e8f7f7286b6ed82585fedc7828fd0c13120800bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b0000bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b0000bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b0000bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee
```

I tested it by passing it to the `tezos-client sign` command

```
$ tezos-client sign bytes '0x03baeb51250afe4344ed1081b8df92a8e8f7f7286b6ed82585fedc7828fd0c13120800bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b0000bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b0000bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b0000bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee' for signatory_test
```

Signatory detected it correctly and signed the operation
```
time="2022-09-29T08:54:44-07:00" level=info msg="GET /authorized_keys" duration="59.575µs" hostname="localhost:6732" method=GET path=/authorized_keys start_time="2022-09-29T08:54:44-07:00" status=200
time="2022-09-29T08:54:44-07:00" level=info msg="Requesting signing operation" ops="map[vdf_revelation:1]" ops_total=1 pkh=tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH request=generic vault=File vault_name=local_file_keys
time="2022-09-29T08:54:44-07:00" level=info msg="About to sign raw bytes" ops="map[vdf_revelation:1]" ops_total=1 pkh=tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH raw=03baeb51250afe4344ed1081b8df92a8e8f7f7286b6ed82585fedc7828fd0c13120800bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b0000bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b0000bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee1250fd0eca7f8b0000bb9d15400eae78d1c8da772a4ffe2296bb72ac4edd02b7f512e80700050194960dffac2498d08551b7dbee request=generic vault=File vault_name=local_file_keys
time="2022-09-29T08:54:44-07:00" level=info msg="Signed generic successfully" ops="map[vdf_revelation:1]" ops_total=1 pkh=tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH request=generic vault=File vault_name=local_file_keys
time="2022-09-29T08:54:44-07:00" level=info msg="POST /keys/tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH" duration="924.386µs" hostname="localhost:6732" method=POST path=/keys/tz1ck3EJwzFpbLVmXVuEn5Ptwzc6Aj14mHSH start_time="2022-09-29T08:54:44-07:00" status=200
```

**Pull request merged**